diff --git a/Makefile.in b/Makefile.in
index 3b3e1f0..562f933 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -29,7 +29,7 @@ endif

 .PHONY: all
 ifeq ($(FLATPAK), xtrue)
-all: umu-dist umu-launcher umu-vendored
+all: umu-dist umu-launcher
 endif

 .PHONY: install
@@ -38,8 +38,8 @@ SOURCE_DATE_EPOCH = $(shell LC_ALL=C date --date='@1580601600')
 all: zipapp
 install: zipapp-install
 else
-all: umu-dist umu-docs umu-launcher umu-vendored umu-delta
-install: umu-install umu-launcher-install umu-vendored-install umu-delta-install
+all: umu-dist umu-docs umu-launcher umu-delta
+install: umu-install umu-launcher-install umu-delta-install
 endif
